PyStemmusScope/docker_process.py

```diff
@@ -0,0 +1,112 @@
+"""The Docker STEMMUS_SCOPE model process wrapper."""
+from PyStemmusScope.config_io import read_config
+from pathlib import Path
+import os
+import docker
+
+
+def make_docker_vols_binds(cfg_file: str) -> tuple[list[str], list[str]]:
+    """Make docker volume mounting configs.
+
+    Args:
+        cfg_file: Location of the config file
+
+    Returns:
+        volumes, binds
+    """
+    cfg = read_config(cfg_file)
+    
+    volumes = [cfg["OutputPath"], cfg["InputPath"]]
+    binds = [
+        f"{cfg['OutputPath']}:{cfg['OutputPath']}:rw",
+        f"{cfg['InputPath']}:{cfg['InputPath']}:ro",
+    ]
+
+    if (
+        not Path(cfg_file).parent.is_relative_to(cfg["InputPath"]) or
+        not Path(cfg_file).parent.is_relative_to(cfg["OutputPath"])
+    ):
+        cfg_folder = str(Path(cfg_file).parent)
+        volumes.append(cfg_folder)
+        binds.append(f"{cfg_folder}:{cfg_folder}:ro")
+
+    return volumes, binds
+
+
+class StemmusScopeDocker:
+    """Communicate with a STEMMUS_SCOPE Docker container."""
+    # The image is hard coded here to ensure compatiblity:
+    image = "ghcr.io/ecoextreml/stemmus_scope:1.5.0"
```

Comment:
Let's not hard-code the path here. Instead, we add the key `model_src_path` to the config file where it is the address to the image or exe file. So, the `ExeFilePath` that is mentioned in the notebook is replaced with `model_src_path`.

Reply:
I did think about this, but it can be hard to ensure compatibility this way and would make it more complex for users :confused: 
At least when it's hard-coded, it's more difficult to break it.

Additionally, there's the problem of how to discern between the path to the model source and a docker container. If you would want to use a local container (e.g. for testing purposes) or one from Docker Hub, then the `ghcr.io` part of the docker image is not there. It can just be `stemmus_scope`.

I before did have an optional DockerImage entry that I added to the config file, which can state which image to use.

Reply:
Perhaps, also in preparation for a grpc4bmi enabled container with the STEMMUS_SCOPE BMI it could work like the following:

1. If ExeFilePath is defined, use that executable file.
2. If DockerImage is defined, use that docker image.
    - We could check the tag for compatibility, and warn if we are not sure if it is compatible with this version of the BMI 
4. If neither is defined, see if STEMMUS_SCOPE is available in the working directory or PATH, and use that.

Reply:
I implemented the steps above. It is now possible to set the STEMMUS_SCOPE environmental variable, which will be the fallback option if the executable file or docker image are not passed.